oci_auth_refresher.sh

```diff
@@ -1,123 +1,240 @@
 #!/bin/zsh
+# shellcheck shell=bash disable=SC1071
 
-# Version: 0.1.0
-# OCI Authentication Refresher
-# This script keeps an OCI session active by refreshing it before it expires
+# ───────────────────────────────────────────────────────────
+# oci_auth_refresher.sh  •  v0.1.1
+#
+# Keeps an OCI CLI session alive by refreshing it shortly
+# before it expires. Intended to be launched (nohup) from the
+# wrapper script oshell.sh.
+# ───────────────────────────────────────────────────────────
 
 # Check if profile argument is provided, use DEFAULT if not
-if [[ -z "$1" ]]
-then
+if [[ -z "$1" ]]; then
   echo "No profile name provided, using DEFAULT"
-  OCI_PROFILE="DEFAULT"
+  OCI_CLI_PROFILE="DEFAULT"
 else
-  OCI_PROFILE=$1
+  OCI_CLI_PROFILE=$1
 fi
 
-# Configuration
-PREEMPT_REFRESH_TIME=60  # Attempt to refresh 60 sec before session expiration
-LOG_LOCATION="${HOME}/Library/Logs/oci-auth-refresher_${OCI_PROFILE}.log"
-SESSION_STATUS_FILE="${HOME}/.oci/sessions/${OCI_PROFILE}/session_status"
+# Check if script is being run directly (not through nohup)
+# If so, relaunch itself using nohup and exit
+if [[ -z "$NOHUP" && -t 1 ]]; then
+  echo "Launching OCI auth refresher in background for profile ${OCI_CLI_PROFILE}"
+  export NOHUP=1
+  # Use full path to script to ensure it's detectable by pgrep
+  script_path=$(cd "$(dirname "${BASH_SOURCE[0]:-$0}")" && pwd)/$(basename "$0")
+  nohup "$script_path" "$OCI_CLI_PROFILE" > /dev/null 2>&1 < /dev/null &
+  pid=$!
+  echo "Process started with PID $pid"
+  echo "You can verify it's running with: pgrep -af oci_auth_refresher.sh"
+  echo "Check logs at: ${HOME}/.oci/sessions/${OCI_CLI_PROFILE}/oci-auth-refresher_${OCI_CLI_PROFILE}.log"
+  exit 0
+fi
+
+# Source common functions from oshell.sh if OSHELL_HOME is set
+if [[ -n "$OSHELL_HOME" && -f "$OSHELL_HOME/oshell.sh" ]]; then
+  # Source the configuration and helper functions
+  # shellcheck disable=SC1090
+  source <(grep -A 100 "^# Configuration" "$OSHELL_HOME/oshell.sh" | grep -B 100 "^function oci_authenticate" | grep -v "^function oci_authenticate")
```

Comment:
It seems like you're moving towards making this script independent of `oshell.sh` which may be fine, but if so, why not source the entire script here?

This line is creating hidden dependencies between this script and `oshell.sh`.

Reply:
Yeah, the partial source with grep does create some coupling. The goal was to keep the script functional even when oshell.sh isn’t available — like when running oci_auth_refresher.sh standalone. That said, I agree it’s not ideal. Open to suggestions, but I think it’s fine for now and something we can revisit and redesign down the road.

Reply:
I think it would make more sense to just source the entire script. Do this wouldn't cause any adverse effects and makes the coupling more durable.

```suggestion
  source "${OSHELL_HOME}/oshell.sh"
```